pajbot/apiwrappers/seventv.py

```diff
@@ -1,73 +1,58 @@
+from typing import Tuple
+
 from pajbot.apiwrappers.base import BaseAPI
 from pajbot.apiwrappers.response_cache import ListSerializer
 from pajbot.models.emote import Emote
 
+from requests import HTTPError
+
 
 class SevenTVAPI(BaseAPI):
     def __init__(self, redis):
-        super().__init__(base_url="https://api.7tv.app/v2/", redis=redis)
+        super().__init__(base_url="https://7tv.io/v3/", redis=redis)
 
     @staticmethod
     def parse_emotes(api_response_data):
-        def get_url(emote_id, size):
-            return f"https://cdn.7tv.app/emote/{emote_id}/{size}x"
+        def get_emote_urls(host) -> Tuple[dict[str, str], int, int]:
+            urls = {}
+            base_url = "https:" + host["url"]
+            emote_size = 1
+            max_width = 0
+            max_height = 0
+            for file in host["files"]:
+                if file["format"] == "WEBP":
+                    name = file["name"]
+                    urls[str(emote_size)] = f"{base_url}/{name}"
+                    if file["width"] > max_width:
+                        max_width = file["width"]
+                    if file["height"] > max_height:
+                        max_height = file["height"]
+                    emote_size += 1
+
+            if len(urls) == 0:
+                raise ValueError("No file in WEBP format for this emote")
```

Comment:
Further exceptions for the max_width or max_height values being zero would be nice

Reply:
Done in e3da46d.